core/src/main/java/org/elasticsearch/action/get/TransportMultiGetAction.java

```diff
@@ -64,12 +64,16 @@ protected void doExecute(final MultiGetRequest request, final ActionListener<Mul
         Map<ShardId, MultiGetShardRequest> shardRequests = new HashMap<>();
         for (int i = 0; i < request.items.size(); i++) {
             MultiGetRequest.Item item = request.items.get(i);
-            if (!clusterState.metaData().hasConcreteIndex(item.index())) {
-                responses.set(i, new MultiGetItemResponse(null, new MultiGetResponse.Failure(item.index(), item.type(), item.id(), new IndexNotFoundException(item.index()))));
+            String concreteSingleIndex;
+            try {
+                // need to resolve date math index expressions. IndicesOptions do not expand wildcards; therefore this method must return
+                // a single index or throw an IndexNotFoundException.
+                concreteSingleIndex = indexNameExpressionResolver.concreteSingleIndex(clusterState, item).getName();
+            } catch (IndexNotFoundException infe) {
```

Comment:
I think catching index not found only is not enough. Other exceptions can be thrown while resolving indices. If you change this to `Exception`, you would also fix #20845 I think. Would you mind  doing that and adding a small test for that to `SimpleMgetIT` ?